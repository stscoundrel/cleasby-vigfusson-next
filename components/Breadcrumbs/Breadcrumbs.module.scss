.section {
  border-bottom: 1px solid $color-blue;
  padding: 16px 0;
}

.link {
  position: relative;
  color: $color-blue;
  text-decoration: none;
  padding: 10px 0;
  padding-right: 15px;
  margin-right: 20px;
  font-size: smaller;
  margin-bottom: 15px;

  &::after {
    content: "/";
    position: absolute;
    right: -5px;
    top: 9px;
  }

  &:last-of-type {
    font-weight: bold;
    text-decoration: underline;
    margin-right: 0;
    padding-right: 0;
    display: block;

    &::after {
      content: "";
    }
  }
}

@media only screen and (min-width: 768px) {
  .link {
    font-size: initial;

    &:last-of-type {
      display: inline;
    }
  }
}