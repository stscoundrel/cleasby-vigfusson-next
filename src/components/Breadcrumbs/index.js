import { getBreadcrumbs } from 'lib/utils/breadcrumbs'
import { getSchema } from 'lib/utils/schema'
import Link from 'next/link'
import styles from './Breadcrumbs.module.scss'

export default function Breadcrumbs({ type, content }) {
  const getBreadCrumbData = () => {
    let letter = null
    let word = null

    if (type !== 'page') {
      letter = type === 'letter' ? content[0].word.charAt(0).toLowerCase() : content.word.charAt(0).toLowerCase()
    }

    if (type === 'word') {
      word = content.word
    }

    return {
      type,
      letter,
      word,
    }
  }

  const breadcrumbs = getBreadcrumbs(getBreadCrumbData())
  const schema = getSchema(breadcrumbs, 'breadcrumbs')

  return (
    <nav className={styles.section}>
      <div className="container">
        {breadcrumbs.map(({ label, url }) => (
          <Link key={url} href={url}>
            <a className={styles.link}>{label}</a>
          </Link>
        ))}

        <script type='application/ld+json' dangerouslySetInnerHTML={ { __html: schema } }/>
      </div>
    </nav>
  )
}
