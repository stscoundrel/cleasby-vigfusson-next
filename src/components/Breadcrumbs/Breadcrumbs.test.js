import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import Breadcrumbs from './index'

describe('Breadcrumbs component', () => {
  test('Default page: does not crach', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<Breadcrumbs type="page" />)
  })

  test('Default page: matches snapshot', () => {
    const tree = renderer.create(<Breadcrumbs type="page" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Default page: renders correct amount of breadcrumbs', () => {
    const tree = renderer.create(<Breadcrumbs type="page" />)
    const { root } = tree

    expect(root.findAllByType('a').length).toEqual(1)
  })

  test('Letter page: does not crach', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<Breadcrumbs type="letter" content={[{ word: 'random' }]} />)
  })

  test('Letter page: matches snapshot', () => {
    const tree = renderer.create(<Breadcrumbs type="letter" content={[{ word: 'random' }]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Letter page: renders correct amount of breadcrumbs', () => {
    const tree = renderer.create(<Breadcrumbs type="letter" content={[{ word: 'random' }]} />)
    const { root } = tree

    expect(root.findAllByType('a').length).toEqual(2)
  })

  test('Word page: does not crach', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<Breadcrumbs type="word" content={{ word: 'random' }} />)
  })

  test('Word page: matches snapshot', () => {
    const tree = renderer.create(<Breadcrumbs type="word" content={{ word: 'random' }} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Word page: renders correct amount of breadcrumbs', () => {
    const tree = renderer.create(<Breadcrumbs type="word" content={{ word: 'random' }} />)
    const { root } = tree

    expect(root.findAllByType('a').length).toEqual(3)
  })
})
