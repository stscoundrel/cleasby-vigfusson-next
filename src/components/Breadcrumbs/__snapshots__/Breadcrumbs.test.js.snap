// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component Default page: matches snapshot 1`] = `
<nav
  className="section"
>
  <div
    className="container"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Cleasby & Vigfusson Dictionary
    </a>
    <script
      dangerouslySetInnerHTML={
        {
          "__html": "{"@context":"https://schema.org/","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":1,"name":"Cleasby & Vigfusson Dictionary","item":"undefined/"}]}",
        }
      }
      type="application/ld+json"
    />
  </div>
</nav>
`;

exports[`Breadcrumbs component Letter page: matches snapshot 1`] = `
<nav
  className="section"
>
  <div
    className="container"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Cleasby & Vigfusson Dictionary
    </a>
    <a
      href="/letter/r"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Letter R
    </a>
    <script
      dangerouslySetInnerHTML={
        {
          "__html": "{"@context":"https://schema.org/","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":1,"name":"Cleasby & Vigfusson Dictionary","item":"undefined/"},{"@type":"ListItem","position":2,"name":"Letter R","item":"undefined/letter/r"}]}",
        }
      }
      type="application/ld+json"
    />
  </div>
</nav>
`;

exports[`Breadcrumbs component Word page: matches snapshot 1`] = `
<nav
  className="section"
>
  <div
    className="container"
  >
    <a
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Cleasby & Vigfusson Dictionary
    </a>
    <a
      href="/letter/r"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Letter R
    </a>
    <a
      href="/word/random"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Random
    </a>
    <script
      dangerouslySetInnerHTML={
        {
          "__html": "{"@context":"https://schema.org/","@type":"BreadcrumbList","itemListElement":[{"@type":"ListItem","position":1,"name":"Cleasby & Vigfusson Dictionary","item":"undefined/"},{"@type":"ListItem","position":2,"name":"Letter R","item":"undefined/letter/r"},{"@type":"ListItem","position":3,"name":"Random","item":"undefined/word/random"}]}",
        }
      }
      type="application/ld+json"
    />
  </div>
</nav>
`;
